src/components/MobileApp/DesignComponents/DatePicker.js

```diff
@@ -0,0 +1,152 @@
+import React, { useEffect, useRef, useState } from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import { format, isAfter, isBefore, isSameDay, isValid, parse } from 'date-fns'
+import { Calendar } from '@leihs/calendar'
+import LabelInside from './LabelInside'
+
+export default function DatePicker({
+  locale,
+  value,
+  onChange,
+  required,
+  className,
+  label,
+  minDate,
+  maxDate,
+  ...restProps
+}) {
+  const formatDate = date => format(date, 'P', { locale: locale })
+  const parseDate = s => parse(s, 'P', new Date(), { locale: locale })
+
+  const containerRef = useRef()
+  const inputRef = useRef()
+
+  const [calendarActive, setCalendarActive] = useState(false)
+  const [wasValidated, setWasValidated] = useState(false)
+  const [validDate, setValidDate] = useState(tryParseDate(value))
+
+  function tryParseDate(stringValue) {
+    if (!stringValue) return null
+    const d = parseDate(stringValue)
+    return isValid(d) ? d : null
+  }
+
+  function handleFocus() {
+    setCalendarActive(true)
+  }
+  function handleBlur() {
+    setWasValidated(true)
+  }
+
+  function handleChange(e) {
+    setValidDate(tryParseDate(e.target.value))
+    onChange(e)
+  }
+
+  function handleCalendarDateChange(d) {
+    setValidDate(d)
+    const stubEvent = { target: { value: formatDate(d) } }
+    onChange(stubEvent)
+    setCalendarActive(false)
+    inputRef.current.focus()
+  }
+
+  function handleClickOutside(e) {
+    if (containerRef.current && !containerRef.current.contains(e.target)) {
+      setCalendarActive(false)
+    }
+  }
+
+  // Propagate outside-triggered value change to calendar
+  useEffect(() => {
+    const validDateFromProp = tryParseDate(value)
+    if (!isSameDay(validDateFromProp, validDate)) {
+      setValidDate(tryParseDate(value))
+    }
+  }, [value])
+
+  // Hide calendar on outside-click
```

Comment:
not sure how well that works in practice, but lets keep it for now. 

* when improving it we should use the same technique/libs like `react-bootstrap` uses. look at the Dropdown and the `autoClose` option it provides. I think many layers below it uses https://react-bootstrap.github.io/react-overlays/api/useRootClose

* also/in general i think clicking the calendar button in the input should also toggle the calendar (it should close it when the calendar is open, the other way already work because `handleFocus`)